modules/ensemble/lib/widget/tooltip.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:ensemble/framework/extensions.dart';
+import 'package:ensemble/widget/calendar.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:ensemble/framework/widget/widget.dart';
+import 'package:ensemble/framework/action.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:ensemble/framework/view/data_scope_widget.dart';
+import 'package:ensemble/page_model.dart';
+import 'package:ensemble/framework/widget/has_children.dart';
+import 'package:ensemble/framework/widget/view_util.dart';
+import 'package:ensemble/framework/device.dart';
+import 'package:yaml/yaml.dart';
+
+
+class ToolTip extends StatefulWidget
+    with Invokable, HasController<ToolTipController, ToolTipState> {
+  static const type = 'ToolTip';
+
+  ToolTip({Key? key}) : super(key: key);
+
+  final ToolTipController _controller = ToolTipController();
+
+  @override
+  ToolTipController get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => ToolTipState();
+
+  @override
+  Map<String, Function> getters() {
+    return {
+      'isVisible': () => _controller.isVisible,
+    };
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'message': (value) => _controller.message = Utils.optionalString(value),
+      'child': (value) => _controller.child = value,
```

Comment:
everywhere else we call it a `widget` suggest calling it a `widget` instead of `child`

Reply:
Updated.